mapsforge-map/src/main/java/org/mapsforge/map/datastore/MapReadResult.java

```diff
@@ -71,14 +75,35 @@ public void add(MapReadResult other, boolean deduplicate) {
                     this.pointOfInterests.add(poi);
                 }
             }
-            for (Way way : other.ways) {
-                if (this.hashWays.add(way.hashCode())) {
-                    this.ways.add(way);
-                }
-            }
```

Comment:
@Sublimis New deduplicate should happen here like the pois.

Reply:
@Sublimis Please see [above](https://github.com/mapsforge/mapsforge/pull/1629#discussion_r1920376921).